native/src/routes/search/components/__tests__/SearchModal.spec.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+
+import React from 'react'
+import { fireEvent, render } from '@testing-library/react-native'
+import SearchModal from '../SearchModal'
+import lightTheme from '../../../../modules/theme/constants'
+import { SEARCH_FINISHED_SIGNAL_NAME } from 'api-client'
+import CategoriesMapModelBuilder from 'api-client/src/testing/CategoriesMapModelBuilder'
+import sendTrackingSignal from '../../../../modules/endpoint/sendTrackingSignal'
+
+jest.mock('rn-fetch-blob')
+jest.mock('../../../../modules/endpoint/sendTrackingSignal')
+
+describe('SearchModal', () => {
+  beforeEach(() => {})
+
+  const dummy = jest.fn()
+  const t = key => key
+  const categoriesMapModel = new CategoriesMapModelBuilder('augsburg', 'de').build()
+
+  it('search modal should send tracking signal on Press', () => {
+    const { getAllByTestId } = render(
```

Comment:
If possible using test ids should be avoided. Instead it would be better to query by e.g. text, see https://testing-library.com/docs/queries/about/#priority

Reply:
Yeah, i coundn't find it byText. Title of the generated Categories must be `Category with id ${id}`, but GetAllByText('Category') does not return anything.

Reply:
I think you need to set `exact` to false as you are just querying by a substring:
https://testing-library.com/docs/queries/about#precision
Did not find docu for native.